src/plugins/vis_builder/public/embeddable/vis_builder_embeddable.tsx

```diff
@@ -296,12 +325,8 @@ export class VisBuilderEmbeddable extends Embeddable<VisBuilderInput, VisBuilder
       dirty = true;
     }
 
-    if (dirty) {
-      this.expression = (await this.getExpression()) ?? '';
-
-      if (this.handler) {
-        this.updateHandler();
-      }
+    if (this.handler && dirty) {
```

Comment:
nit: Why do we need this change? Feels like the original logic was intentional.

Reply:
don't remember why but it is a request change. otherwise will see
<img width="1121" alt="Screenshot 2024-04-29 at 1 15 05 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/8580c20d-2bdc-4232-bdab-4b5a4839de72">